spec/suites/layer/GeoJSONSpec.js

```diff
@@ -512,3 +512,387 @@ describe("L.LayerGroup#toGeoJSON", function () {
+				type: "Feature",
+				geometry: point,
+				properties: {radius: 100}
+			}, {
+				pointToLayer: customPointToLayer
+			});
+			expect(layer instanceof L.Circle).to.be(true);
+			expect(layer.options.radius).to.be(100);
+		});
+
+		it("creates a Layer using pointToLayer option (MultiPoint)", function () {
+			const layer = L.GeoJSON.geometryToLayer({
+				type: "Feature",
+				geometry: multiPoint,
+				properties: {radius: 100}
+			}, {
+				pointToLayer: customPointToLayer
+			});
+			Object.keys(layer._layers).forEach(function (id) {
+				expect(layer._layers[id] instanceof L.Circle).to.be(true);
+				expect(layer._layers[id].options.radius).to.be(100);
+			});
+		});
+
+		it("creates a Layer using coordsToLatLng option (Point)", function () {
+			const layer = L.GeoJSON.geometryToLayer({
+				type: "Feature",
+				geometry: {
+					type: "Point",
+					coordinates: [1, 2, 3]
+				}
+			}, {
+				coordsToLatLng: customCoordstoLatLng
+			});
+			expect(layer._latlng).to.eql({lat: 3, lng: 2, alt: 4});
+		});
+
+		it("creates a Layer using coordsToLatLng option (MultiPoint)", function () {
+			const layer = L.GeoJSON.geometryToLayer({
+				type: "Feature",
+				geometry: {
+					type: "MultiPoint",
+					coordinates: [
+						[1, 2, 3], [4, 5, 6]
+					]
+				}
+			}, {
+				coordsToLatLng: customCoordstoLatLng
+			});
+			const expected = [
+				{lat: 3, lng: 2, alt: 4},
+				{lat: 6, lng: 5, alt: 7}
+			];
+			const ids = Object.keys(layer._layers);
+			for (var i = 0; i < ids.length; i++) {
+				expect(layer._layers[ids[i]]._latlng).to.eql(expected[i]);
+			}
+		});
+	});
+
+	describe("#coordsToLatLng", function () {
+		it("creates a LatLng object given longitude and latitude", function () {
+			const latlng = L.GeoJSON.coordsToLatLng([0, 0]);
+			expect(latlng instanceof L.LatLng).to.be(true);
+			expect(latlng.lat).to.be(0);
+			expect(latlng.lat).to.be(0);
+			expect(latlng.alt).to.be(undefined);
+		});
+
+		it("creates a LatLng object given longitude, latitude and altitude", function () {
+			const latlng = L.GeoJSON.coordsToLatLng([0, 0, 0]);
+			expect(latlng instanceof L.LatLng).to.be(true);
+			expect(latlng.lat).to.be(0);
+			expect(latlng.lat).to.be(0);
+			expect(latlng.alt).to.be(0);
+		});
+	});
+
+	describe("#coordsToLatLngs", function () {
+
+		function customCoordsToLatLng(coords) {
+			return new L.LatLng(coords[0] + 1, coords[1] + 1, coords[2] + 1);
+		}
+
+		it("creates a multidimensional array of LatLngs", function () {
+			const latLngs = L.GeoJSON.coordsToLatLngs([[0, 0], [1, 1], [2, 2]]);
+			expect(latLngs.length).to.be(3);
+			latLngs.forEach(function (latLng) {
+				expect(latLng instanceof L.LatLng).to.be(true);
+			});
+		});
+
+		it("creates a multidimensional array of LatLngs (levelsDeep=1)", function () {
+			const latLngs = L.GeoJSON.coordsToLatLngs([
+				[[0, 0], [1, 1], [2, 2]],
+				[[4, 4], [5, 5], [6, 6]]
+			], 1);
+			expect(latLngs.length).to.be(2);
+			for (var i = 0; i < latLngs.length; i++) {
```

Comment:
Keep it simpler, avoid `for` where you can use `forEach`.
May be even https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat

Reply:
tried flat, doesnt work. Using forEach two times now.